src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -197,6 +200,48 @@ public CaseDTO getCaseById(final UUID caseId, CaseQueryRequestDTO requestParamsD
     return mapCaseToDtoList(dbCases);
   }
 
+  @Override
+  public List<CaseSummaryDTO> getCaseSummaryBySampleAttribute(String key, String value)
+      throws CTPException {
+    if (log.isDebugEnabled()) {
+      log.debug(
+          "Fetching latest case summary details by {}", key, kv("key", key), kv("value", value));
+    }
+
+    // Find matching cases
+    List<Case> dbCases;
+    try {
+      dbCases = getCasesFromDb(key, value);
+    } catch (CTPException ex) {
+      if (ex.getFault() == Fault.RESOURCE_NOT_FOUND) {
+        log.info(
+            "Case by {} Not Found calling Case Service", key, kv("key", key), kv("value", value));
+        return Collections.emptyList();
+      } else {
+        log.error("Error calling Case Service", kv("key", key), kv("value", value), ex);
+        throw ex;
+      }
+    }
+
+    // Summarise all found cases
+    List<CaseSummaryDTO> caseSummaries = new ArrayList<>();
+    for (Case dbCase : dbCases) {
+      CaseSummaryDTO caseSummary = new CaseSummaryDTO();
+      caseSummary.setId(dbCase.getId());
+      caseSummary.setCaseRef(dbCase.getCaseRef());
+
+      Survey survey = dbCase.getCollectionExercise().getSurvey();
+      caseSummary.setSurveyName(survey.getName());
+
+      SurveyType surveyType = SurveyType.fromSampleDefinitionUrl(survey.getSampleDefinitionUrl());
+      caseSummary.setSurveyType(surveyType.getBasename());
```

Comment:
The baseName is the SurveyTypes internal clue as to which one it is.
TBH not even sure getBasename() should be public to avoid this again.
We should return SurveyType.name() when we send it outbound ie 'SOCIAL'
ie what if the enum is refactored and we refer on hand to SOCIAL_SURVEY and on the other, the internal 'social'?

Reply:
Have changed to SurveyType.name()